ts/src/hitbtc.ts

```diff
@@ -2131,18 +2144,15 @@ export default class hitbtc extends Exchange {
         } else if ((type === 'stopLimit') || (type === 'stopMarket') || (type === 'takeProfitLimit') || (type === 'takeProfitMarket')) {
             throw new ExchangeError (this.id + ' createOrder() requires a stopPrice parameter for stop-loss and take-profit orders');
         }
-        let marketType = undefined;
-        [ marketType, params ] = this.handleMarketTypeAndParams ('createOrder', market, params);
         let method = this.getSupportedMapping (marketType, {
             'spot': 'privatePostSpotOrder',
             'swap': 'privatePostFuturesOrder',
             'margin': 'privatePostMarginOrder',
         });
-        const [ marginMode, query ] = this.handleMarginModeAndParams ('createOrder', params);
         if (marginMode !== undefined) {
             method = 'privatePostMarginOrder';
         }
-        params = this.omit (params, [ 'triggerPrice', 'timeInForce', 'time_in_force', 'stopPrice', 'stop_price', 'reduceOnly' ]);
+        params = this.omit (params, [ 'triggerPrice', 'timeInForce', 'stopPrice', 'stop_price', 'reduceOnly', 'postOnly' ]);
         const response = await this[method] (this.extend (request, query));
```

Comment:
@samgermain this logic is wrong, we omit from params and then use `query`

Reply:
> @samgermain this logic is wrong, we omit from params and then use `query`

you're right, sorry